rasterio/plot.py

```diff
@@ -130,17 +134,30 @@ def show(source, with_bounds=True, contour=False, contour_label_kws=None,
     return ax
 
 
-def plotting_extent(source):
+def plotting_extent(source, transform=None):
     """Returns an extent in the format needed
      for matplotlib's imshow (left, right, bottom, top)
      instead of rasterio's bounds (left, bottom, top, right)
 
     Parameters
     ----------
-    source : raster dataset
+    source : raster dataset or array
+    transform: Affine, required if source is array
     """
-    extent = (source.bounds.left, source.bounds.right,
-              source.bounds.bottom, source.bounds.top)
+    try:
+        extent = (source.bounds.left, source.bounds.right,
+                  source.bounds.bottom, source.bounds.top)
+    except AttributeError:
+        # no bounds attribute, try array
+        if transform is None:
+            raise ValueError(
+                "transform is required if source is an array")
+        transform = guard_transform(transform)
+        rows, cols = source.shape[0:2]
```

Comment:
@perrygeo these be the last two indices rather than first two, right?

`rasterio.open('tests/data/RGB.byte.tif').read().shape` is `(3, 718, 791)` and we want `(718, 791)` here.